src/app/core/services/authenticate.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ApiService, AuthToken, UserDTO } from '../../../backend-client';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticateService {
+  private currentUserSubject: BehaviorSubject<Observable<UserDTO> | null>;
+  public currentUser: Observable<Observable<UserDTO> | null>;
```

Comment:
I think it's common to add dollar sign ($) to observables because it's easy to distinguish between typical variable and Observables. So this  would become `currentUser$` you can discuss with others about this @new5558

Reply:
https://medium.com/@benlesh/observables-are-just-functions-but-also-collections-how-do-i-name-them-918c5ce2f64 

I think. It's a good idea to use "$".

Reply:
agree krub